### DISCUSSION TITLE
How to eliminate libraries from the class path

### DISCUSSION THREAD

**Discussion by kastourik12** [burningwave/core#36] [Q&A]
Hello guys, I am trying to use this library (I am using the static instances )and I want to find the classes that are annotated with a specific annotation or if they have members who are annotated with the same annotation the problem is that this is also loading the classes from the libraries which I want to avoid. is there a way to do it, or specify to only look into the main package, Thanks in advance.

**Comment by Roberto-Gentili** [comment]
> Hello guys, I am trying to use this library (I am using the static instances )and I want to find the classes that are annotated with a specific annotation or if they have members who are annotated with the same annotation the problem is that this is also loading the classes from the libraries which I want to avoid. is there a way to do it, or specify to only look into the main package, Thanks in advance.

You can use a file filter (check the second example [here](https://github.com/burningwave/core/wiki/How-to-search-for-all-classes-that-have-package-name-that-matches-a-regex)) in combination with the ClassCriteria [to search for a specific annotation](https://github.com/burningwave/core/wiki/Finding-all-annotated-classes).
You can also [reduce the classpaths where to search by passing them directly](https://github.com/burningwave/core/wiki/How-to-retrieve-all-classes-of-the-classpath) or use a custom or [an isolated class loader](https://github.com/burningwave/core/blob/bb88e19a7ae28a99add46a5915660596b48842ab/src/test/java/org/burningwave/core/ClassHunterTest.java#L865) that you will discard once you finished the search
